schema/proto3/neighbor.proto

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2019 The Alcor Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+        you may not use this file except in compliance with the License.
+        You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+
+syntax = "proto3";
+
+package alcor.schema;
+
+option java_package = "com.futurewei.alcor.schema";
+option java_outer_classname = "Neighbor";
+
+import "common.proto";
+
+enum NeighborType { 
+    L2 = 0; // the default type
+    L3 = 1;
+    HOST_DVR_MAC = 2;
+}
+
+message NeighborConfiguration {
+    uint32 format_version = 1;
+    uint32 revision_number = 2;
+
+    string id = 3;
+    NetworkType network_type = 4;  // we may move this to subnet level
```

Comment:
I see Subnet already has network_type field so we could remove it from PortConfiguration and NeighborConfiguration.

Reply:
removed in PortConfiguration and NeighborConfiguration. The removal in PortConfiguration is a breaking change for ACA, but I will deal with it.

Reply:
the removal in PortConfiguration turned out to be a breaking change in DPM @haboy52581 . I will keep it in PortConfiguration for now.